src/scripts/Joyride.jsx

```diff
@@ -134,8 +138,11 @@ class Joyride extends React.Component {
   componentWillReceiveProps(nextProps) {
     const { play, shouldPlay, standaloneTooltip } = this.state;
     const { keyboardNavigation, run } = this.props;
-    this.logger('joyride:willReceiveProps', [nextProps]);
-
+    logger({
+      type: 'joyride:willReceiveProps',
+      msg: [nextProps],
+      debug: nextProps.debug,
```

Comment:
Note that this is looking at the debug of `nextProps`, whereas before the method would have looked at `this.props`.  I think this is correct, but wanted to call it out.